fastlane/lanes/release.rb

```diff
@@ -158,34 +288,52 @@
   #####################################################################################
   desc 'Updates store metadata and runs the release checks'
   lane :finalize_release do |options|
-    UI.user_error!('Please use `finalize_hotfix_release` lane for hotfixes') if android_current_branch_is_hotfix()
+    UI.user_error!('Please use `finalize_hotfix_release` lane for hotfixes') if android_current_branch_is_hotfix
+
+    ensure_git_status_clean
+    ensure_git_branch(branch: '^release/') # Match branch names that begin with `release/`
+
+    message = "Finalizing release: #{current_release_version}\n"
+    if options[:skip_confirm]
+      UI.message(message)
+    else
+      UI.user_error!('Aborted by user request') unless UI.confirm("#{message}Do you want to continue?")
+    end
 
-    android_finalize_prechecks(skip_confirm: options[:skip_confirm])
     configure_apply(force: is_ci)
 
-    app_version = android_get_app_version()
+    app_version = current_release_version
     release_branch = "release/#{app_version}"
 
     # Remove branch protection first, so that we can push the final commits directly to the release branch
     removebranchprotection(repository: GHHELPER_REPO, branch: release_branch)
 
     # Don't check translation coverage for now since we are finalizing the release in CI
-    # check_translations_coverage()
-    download_translations()
+    # check_translations_coverage
+    download_translations
+
+    # Bump the release version and build code
+    UI.message 'Bumping final release version and build code...'
+    VERSION_FILE.write_version(
+      version_name: current_release_version,
+      version_code: next_build_code
+    )
+    commit_version_bump
+    UI.message "Done! New Release Version: #{current_release_version}. New Build Code: #{current_build_code}"
 
-    android_bump_version_final_release()
-    version = android_get_release_version()
-    download_metadata_strings(version: version['name'], build_number: version['code'])
+    version = current_release_version
```

Comment:
Should we rename this to `version_name` to differentiate it from the previous `version` model?

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-Android/commit/6ab18efe9261a63b77766614c06c3bc661a697f6